### PR TITLE
Add Redirect class

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

File: src/Redirect.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import RedirectException from './RedirectException';
+
+export default class Redirect {
+  constructor({ from, to }) {
```
Comment (taion) [inline src/Redirect.js]: I think this reads a little bit better than using positional args. See examples/basic.

File: src/createBaseRouter.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ export default function createBaseRouter({ routeConfig }) {
     }) {
       const currentMatchIndex = this.matchIndex;
       const routes = getRoutes(routeConfig, match);
-      const fullMatch = { ...match, routes, context: matchContext };
+      const fullMatch = { ...match, routes, matcher, context: matchContext };
```
Comment (taion) [inline src/createBaseRouter.js]: `getComponent`, `getData`, and `render` now have access to `matcher`. Should be okay, right?

File: src/Matcher.js [hunk 3]
```diff
@@ -113,4 +113,8 @@ export default class Matcher {
         isEqual(matchQuery[key], value)
     ));
   }
+
+  format(pattern, params) {
```
Comment (taion) [inline src/Matcher.js]: reasonable name?

File: src/Redirect.js (hunk 1 above)
Comment (taion) [inline src/Redirect.js]: `path` or `from`?

Comment (jquense) [inline src/Redirect.js]: the `new` context i like `from` better personally

Comment (taion) [inline src/Redirect.js]: i.e. this?

**Comment by taion** [comment]
Okay to merge?

**Comment by jquense** [comment]
👍

**Pull request merged**